lib/algos/list/delimiter.js

```diff
@@ -148,10 +148,18 @@ class Delimiter extends Extension {
     filter(obj) {
         const key = obj.key;
         const value = obj.value;
-        if ((this.prefix && !key.startsWith(this.prefix))
-            || (this.alphabeticalOrder
-                && typeof this.NextMarker === 'string'
-                && key <= this.NextMarker)) {
+        if (this.prefix && !key.startsWith(this.prefix)) {
```

Comment:
you mention we assume we work with backends that are alphabetically ordered. If possible and if it makes sense here, can we add  `this.aphabeticalOrder`. Would be nice here to make it more apparent

Reply:
Sure, the point was that I have no idea if this alphabeticalOrder flag is actually necessary or not, given that I have no idea why we'd not be using it.
I'll add the check.